### ISSUE TITLE
Using Joken.Plug with Phoenix causes a compile-time error

### ISSUE THREAD

**Issue by mattweldon** [bryanjos/joken#102]
Just tried to upgrade from `plug_jwt` / `joken 0.15.0` to `joken 0.16.0` using the upgrade guides [here](https://github.com/bryanjos/joken/blob/13b8212d189cb6d754166bd697ff5a892e8d8362/CHANGELOG.md), and now I can't get my Phoenix apps to compile.

The error returned is:

```
== Compilation error on file web/router.ex ==
** (ArgumentError) cannot escape #Function<0.82266480/2 in Joken.Plug.init/1>. The supported values are: lists, tuples, maps, atoms, numbers, bitstrings, pids and remote functions in the format &Mod.fun/arity
    (elixir) src/elixir_quote.erl:302: :elixir_quote.bad_escape/1
    (elixir) src/elixir_quote.erl:246: :elixir_quote.do_quote/3
    (elixir) src/elixir_quote.erl:141: :elixir_quote.escape/2
    (elixir) lib/macro.ex:307: Macro.escape/2
    (plug) lib/plug/builder.ex:240: Plug.Builder.quote_plug_call/3
    (plug) lib/plug/builder.ex:215: Plug.Builder.quote_plug/4
    (elixir) lib/enum.ex:1261: Enum."-reduce/3-lists^foldl/2-0-"/3
    (plug) lib/plug/builder.ex:186: Plug.Builder.compile/3
    web/router.ex:12: (module)
    (stdlib) erl_eval.erl:669: :erl_eval.do_apply/6
    (elixir) lib/kernel/parallel_compiler.ex:97: anonymous fn/4 in Kernel.ParallelCompiler.spawn_compilers/8
```

Unsure if this is a compatibility problem with Joken/Phoenix, or if I've just missed something obvious!

I've managed to replicate the issue in a [vanilla Phoenix app](https://github.com/mattweldon/phoenix_joken_auth) running `Elixir 1.0.5` *and* `1.1.1`

**Comment by bryanjos** [comment]
Thanks for the project with the issue replicated. I was able to track it down and fix it. I'll update the version in hex shortly and let you know.

**Comment by bryanjos** [comment]
@mattweldon I published the new version (0.16.1) in hex. Update the joken dependency and try again. Hopefully it will work with no further issues.

**Comment by mattweldon** [comment]
@bryanjos Worked a charm! Thanks :+1:
